### DISCUSSION TITLE
Blank page when using Zerotier

### DISCUSSION THREAD

**Discussion by qingfua** [coder/code-server#2608] [General]
- Web Browser: Chrome
- Local OS: Windows
- Remote OS: Ubuntu 20.04.1 LTS
- Remote Architecture: x86_64
- `code-server --version`: 3.6.0 a4a03c14922ccaec2a9ff8d1b7b2af8522a4214d

I use Zerotier to build a local network, and changed the bind-addr config to local IP
![image](https://user-images.githubusercontent.com/16662647/95946603-784cb480-0e1f-11eb-88e3-5005b986ad1f.png)

The auth page is perfect
![image](https://user-images.githubusercontent.com/16662647/95946642-8e5a7500-0e1f-11eb-837d-f2dd29d3bd9f.png)

But the editor is a blank page
![image](https://user-images.githubusercontent.com/16662647/95947003-4b4cd180-0e20-11eb-8f10-59c5c454f163.png)

The loading spinner wouldn't disappear, but the network tab in develop tools showed finished
![image](https://user-images.githubusercontent.com/16662647/95946841-f9a44700-0e1f-11eb-9839-b831ca74d4eb.png)
![image](https://user-images.githubusercontent.com/16662647/95946878-0d4fad80-0e20-11eb-805f-0d20ada390fa.png)

And the console also has no error
![image](https://user-images.githubusercontent.com/16662647/95946991-438d2d00-0e20-11eb-86c8-c2c048732b94.png)

So strange, how can I solve it?

**Comment by code-asher** [comment]
Usually the blank page is because the web sockets aren't connecting. Can
you run code-server with `--log trace` and see if anything shows up in
the browser console or the terminal?

**Comment by qingfua** [comment]
@code-asher I found if i wait enough long time, there are some error indeed, here are the terminal output: https://paste.ubuntu.com/p/Cd8ktXWP79/
And borwser console are here: https://paste.ubuntu.com/p/RFjqs6WjfY/
![image](https://user-images.githubusercontent.com/16662647/96110688-cabacd80-0f12-11eb-956b-750e120dcc5f.png)

**Comment by code-asher** [comment]
Interesting, the 404'd files are ones that should only load during
development. Have you ever ran code-server from the git repository
before? Or maybe it's falling back to them because the main workbench
file has failed to load. Either way, I think they're a symptom rather
than the cause.

I think the connection reset seems likely to be the main problem and I'm
not really sure how to go about debugging that but it seems likely to be
network-related. Maybe it's has something to do with the size of the
file. Are you able to load `workbench.web.api.js` if you browse to it
directly?

**Comment by qingfua** [comment]
> Interesting, the 404'd files are ones that should only load during development. Have you ever ran code-server from the git repository before? Or maybe it's falling back to them because the main workbench file has failed to load. Either way, I think they're a symptom rather than the cause. I think the connection reset seems likely to be the main problem and I'm not really sure how to go about debugging that but it seems likely to be network-related. Maybe it's has something to do with the size of the file. Are you able to load `workbench.web.api.js` if you browse to it directly?

The `workbench.web.api.js` showed unfinished in console, and it also showed a part when browsing directly. I guess it maybe caused by the bandwidth of 1Mbps, but if I bind the IP to public network, it will be normal.

**Comment by reesericci** [comment]
@zz5840 I find it works for me when I bind to 0.0.0.0, but I don't know if that is practical in your setup.

**Comment by qingfua** [comment]
@reesericci Did you bind to 0.0.0.0 and visit via the local ip of zerotire?

**Comment by reesericci** [comment]
Yea

**Comment by qingfua** [comment]
> Yea

I will try it later, thanks

**Comment by nhooyr** [comment]
There's no evidence of a code-server bug here so I converted this to a discussion.
